oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtAuthenticatedPrincipal.java

```diff
@@ -14,70 +14,75 @@
  * limitations under the License.
  */
 
-package org.springframework.security.oauth2.core;
+package org.springframework.security.oauth2.jwt;
+
+import static org.springframework.security.core.authority.AuthorityUtils.NO_AUTHORITIES;
 
 import java.io.Serializable;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Map;
 
 import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.core.OAuth2AuthenticatedPrincipal;
 import org.springframework.util.Assert;
 
-import static org.springframework.security.core.authority.AuthorityUtils.NO_AUTHORITIES;
-
 /**
- * A domain object that wraps the attributes of an OAuth 2.0 token.
+ * A domain object that wraps the attributes of OAuth 2.0 JSON Web Token (JWT).
  *
- * @author Clement Ng
- * @author Josh Cummings
- * @since 5.2
+ * @author David Kovac
+ * @since 5.4
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7519#section-4.1">Registered Claim Names</a>
  */
-public final class DefaultOAuth2AuthenticatedPrincipal implements OAuth2AuthenticatedPrincipal, Serializable {
+public final class JwtAuthenticatedPrincipal implements OAuth2AuthenticatedPrincipal, JwtClaimAccessor, Serializable {
```

Comment:
Additionally, `OAuth2IntrospectionAuthenticatedPrincipal` only makes sense when OAuth 2.0 Introspection was performed to retrieve the token attributes. For applications that introspected the token, but not with OAuth 2.0 Introspection, they can use `DefaultOAuth2AuthenticatedPrincipal` for additional flexibility.

Reply:
I mentioned it in issue that I was not sure if it can be removed (because it is a public class), but introspecting token without OAuth2 introspection is a good point